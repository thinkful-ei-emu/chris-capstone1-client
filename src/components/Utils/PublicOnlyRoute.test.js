import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PublicOnlyRoute from './PublicOnlyRoute';
import ReactDom from 'react-dom';
import { BrowserRouter } from 'react-router-dom';

describe('PublicOnlyRoute component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(
          <BrowserRouter>
            <PublicOnlyRoute
            exact path ={'/'}
            component={LandingPage} 
            />
          </BrowserRouter>,
          div
        )
        ReactDom.unmountComponentAtNode(div)
      })
    it('renderes PublicOnlyRoute by default', () => {
        expect(toJson(shallow(<PublicOnlyRoute
          exact path ={'/'}
          component={LandingPage} 
           />))).toMatchSnapshot() 
    })
})

export default class LandingPage extends React.Component {

  render() {
      return (
          <section></section>
      )
  }
}