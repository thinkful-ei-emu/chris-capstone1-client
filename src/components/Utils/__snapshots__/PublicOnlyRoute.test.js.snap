// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicOnlyRoute component renderes PublicOnlyRoute by default 1`] = `
<Route
  exact={true}
  path="/"
  render={[Function]}
/>
`;
